### ISSUE TITLE
Streamlining function evaluation for simple cases

### ISSUE THREAD

**Issue by RemDelaporteMathurin** [scientificcomputing/scifem#56]
While I understand the method presented [here](https://jsdokken.com/FEniCS-workshop/src/deep_dive/expressions.html#) provides enough flexibility and is very robust for parallel runs, it is a bit cumbersome and lengthy and I wondered if we could provide a convenience function for simple cases.

In legacy-fenics, we could evaluate a function simply by `u.eval((0, 0, 0))`. This was a bit fragile in parallel of course.

Could we imagine a convenience function to add this?

**Comment by finsberg** [comment]
I agree this would be nice to have. Maybe we could have 
```python
def evaluate_function(u: dolfinx.fem.Function, points: npt.NDArray[np.float64]): ...
```
I think it would also be nice to have a convenient way to evaluate `ufl` expressions, provided the user supplies an appropriate function space, i.e
```python
def evaluate_expression(u: ufl.core.expr.Expr, V: dolfinx.fem.FunctionSpace, points: npt.NDArray[np.float64]): ...
```
To handle issues with parallelism we could also add a argument `broadcast=True` that would ensure that values are present on all processors, i.e 
```python
def evaluate_function(u: dolfinx.fem.Function, points: npt.NDArray[np.float64], broadcast: bool = True): ...
```

**Comment by RemDelaporteMathurin** [comment]
@finsberg that would be lovely. I really like the idea of evaluation ufl expressions!!!

Not sure I understand the `broadcast` argument here. If True, it would run an internal check and return an error if the point is not on all procs?

**Comment by finsberg** [comment]
@RemDelaporteMathurin This has now been resolved for function evaluation here: https://github.com/scientificcomputing/scifem/pull/74
